src/contracts/interfaces/IBLSPubkeyRegistry.sol

```diff
@@ -20,44 +20,61 @@ interface IBLSPubkeyRegistry is IRegistry {
-     * @dev Permissioned by RegistryCoordinator
-     */    
-    function deregisterOperator(address operator, uint8[] memory quorumNumbers, BN254.G1Point memory pubkey) external returns(bytes32);
+     * @notice Deregisters the `operator`'s pubkey for the specified `quorumNumbers`.
+     * @param operator The address of the operator to deregister.
+     * @param quorumNumbers The quourm numbers the operator is deregistering from, where each byte is an 8 bit integer quorumNumber.
+     * @param pubkey The public key of the operator.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions:
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already deregistered
+     *         5) `quorumNumbers` is the same as the parameter use when registering
+     *         6) `pubkey` is the same as the parameter used when registering
+     */ 
+    function deregisterOperator(address operator, bytes memory quorumNumbers, BN254.G1Point memory pubkey) external returns(bytes32);
 
-    /// @notice returns the current stored APK for the `quorumNumber`
-    function quorumApk(uint8 quorumNumber) external view returns (BN254.G1Point memory);
+    /// @notice Returns the current APK for the provided `quorumNumber `
+    function getApkForQuorum(uint8 quorumNumber) external view returns (BN254.G1Point memory);
 
-    /// @notice returns the current stored APK among all quorums
-    function globalApk() external view returns (BN254.G1Point memory);
-
-    /// @notice returns the `ApkUpdate` struct at `index` in the list of APK updates for the `quorumNumber`
+    /// @notice Returns the `ApkUpdate` struct at `index` in the list of APK updates for the `quorumNumber`
     function getApkUpdateForQuorumByIndex(uint8 quorumNumber, uint256 index) external view returns (ApkUpdate memory);
 
-    /// @notice returns the `ApkUpdate` struct at `index` in the list of APK updates that keep track of the sum of the APK amonng all quorums
-    function globalApkUpdates(uint256 index) external view returns (ApkUpdate memory);
-
     /**
      * @notice get hash of the apk of `quorumNumber` at `blockNumber` using the provided `index`;
      * called by checkSignatures in BLSSignatureChecker.sol.
+     * @param quorumNumber is the quorum whose ApkHash is being retrieved
+     * @param blockNumber is the number of the block for which the latest ApkHash muust be retrieved
+     * @param index is the provided witness of the onchain index calculated offchain
      */
-    function getApkHashForQuorumAtBlockNumberFromIndex(uint8 quorumNumber, uint32 blockNumber, uint256 index) external view returns (bytes32);
+    function getApkHashForQuorumAtBlockNumberFromIndex(uint8 quorumNumber, uint32 blockNumber, uint256 index) external returns (bytes32);
```

Comment:
I'm assuming the change from 'view' to non-view is transient / for logging purposes?

Reply:
yup